hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactingToCellArrayMapMemStore.java

```diff
@@ -264,7 +265,7 @@ public void testFlattening() throws IOException {
     for ( Segment s : memstore.getSegments()) {
       counter += s.getCellsCount();
     }
-    assertEquals(10,counter);
+    assertEquals(16,counter);
```

Comment:
so this was 10 because of compaction and now it is 16 because of merge.
do you have a test that covers the compaction scenario? or can we set this test to do merge and then do compaction and see that it is reduced to 10?

DONE review!